src/fw-logs/fw-log-data.cpp

```diff
@@ -1,68 +1,69 @@
 // License: Apache 2.0. See LICENSE file in root directory.
-// Copyright(c) 2019 Intel Corporation. All Rights Reserved.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
 #include "fw-log-data.h"
-#include <sstream>
-#include <iomanip>
-#include <locale>
-#include <string>
 
-using namespace std;
+#include <rsutils/string/from.h>
 
-# define SET_WIDTH_AND_FILL(num, element) \
-setfill(' ') << setw(num) << left << element \
 
 namespace librealsense
 {
     namespace fw_logs
     {
-        rs2_log_severity fw_log_data::get_severity() const
-        {
-            return fw_logs_severity_to_log_severity(_severity);
-        }
-
-        const std::string& fw_log_data::get_message() const
-        {
-            return _message;
-        }
-
-        const std::string& fw_log_data::get_file_name() const
-        {
-            return _file_name;
-        }
-
-        const std::string& fw_log_data::get_thread_name() const
+        rs2_log_severity fw_logs_binary_data::get_severity() const
         {
-            return _thread_name;
-        }
+            const fw_log_binary * log_binary = reinterpret_cast< const fw_log_binary * >( logs_buffer.data() );
```

Comment:
No need to verify not NULL?
static analysis will not yell here?

Reply:
Same question for all other function dereference pointers in this file.

Reply:
Static analysis did not complain, it can cast `nullptr` pointer type.

The buffer data is [populated before pushing to device queue](https://github.com/IntelRealSense/librealsense/pull/12688/files#diff-74e5a7e46fd22ad9f2e85ae20ff9f90c55e2e31bdab4c9b407b7362f2ba7449aR97), but theoretically, there may be a problem when accessing the `severity` field. I will add a check.

Reply:
Done